plugin/pulpcore/plugin/download/http.py

```diff
@@ -118,10 +148,14 @@ def __init__(self, url, session=None, auth=None, proxy=None, proxy_auth=None,
                               url=self.url, exception=None)
 
     @attach_url_to_exception
+    @backoff.on_exception(backoff.expo, aiohttp.ClientError, max_tries=10, giveup=inspect_to_retry)
```

Comment:
Since not all ```aiohttp.ClientError``` have a status code, either this code should be made more specific by specifying ```aiohttp.ClientResponseError``` or the function above should account for the fact that not all subclasses of ```aiohttp.ClientError``` have status codes since they aren't all http-related.  Potentially some of those would still warrant a retry but I haven't really looked thoroughly (```ServerTimeoutError```? ```ServerDisconnectedError```?  ```ClientConnectionError```?)

Reply:
I see exactly what you are saying, and I agree.

I only intend to capture HTTP 429's for now. That's the only error code we have a concrete use case for. That exception type will only ever come as a the `aiohttp.ClientResponseError` so I'm updating the type to be that. This also matches the `exc.code` change from the above comment.